File: tools/sizing/defScalingFactor.m
function Factor = defScalingFactor(k_radial, k_axial, defineType, varargin)
    Factor.k_Axial = k_axial;  % Axial 방향 스케일링 팩터
    Factor.k_Radial = k_radial;  % Radial 방향 스케일링 팩터

    if defineType == 0
        if nargin < 4
            error('defineType 0에 대한 충분한 입력 인수가 없습니다.');
        end
        Factor.k_Winding = varargin{1};  % Winding 스케일링 팩터
    elseif defineType == 1
        if nargin < 5
            error('defineType 1에 대한 충분한 입력 인수가 없습니다.');
        end
        Factor.turns_per_coil = varargin{1};  % 턴당 코일 수
        Factor.a_p = varargin{2};  % a_p (병렬 경로)
    elseif defineType == 2 % 전체턴수 입력
        if nargin < 7
            error('defineType 2에 대한 충분한 입력 인수가 없습니다.');
        end
        Factor.turns_per_coil = varargin{1};  % 턴당 코일 수
        Factor.a_p = varargin{2};  % a_p (병렬 경로)
        n_c_ref = varargin{3};  % n_c_ref (참조값)
        a_p_ref = varargin{4};  % a_p_ref (참조값)
        Factor.k_Winding = (Factor.turns_per_coil / Factor.a_p) / (n_c_ref / a_p_ref);  % Winding 스케일링 팩터 계산
    elseif defineType == 3
        if nargin < 4
            error('defineType 3에 대한 충분한 입력 인수가 없습니다.');
        end
        Factor.turns_per_coil = varargin{1};  % 턴당 코일 수
        Factor.a_p = varargin{2};  % a_p (병렬 경로)
        MotorCADGeo = varargin{3};  % MotorCADGeo 데이터 입력

        if MotorCADGeo.Armature_CoilStyle == 0
            n_c_ref = double(MotorCADGeo.MagTurnsConductor); % turn per Coil
        elseif MotorCADGeo.Armature_CoilStyle == 1
            n_c_ref = MotorCADGeo.WindingLayers;
        end

        a_p_ref = double(MotorCADGeo.ParallelPaths); % Parallel Path
        Factor.k_Winding = (Factor.turns_per_coil / Factor.a_p) / (n_c_ref / a_p_ref);  % Winding 스케일링 팩터 계산
    else
        error('잘못된 defineType입니다. 0, 1, 2 또는 3이어야 합니다.');
    end
end

File: tools/motorCAD/pyMCAD/scaling/def_scaling_factor.py
"""
def_scaling_factor.py — MATLAB defScalingFactor 대응

사용 예 (JEET):
    defScalingFactor(2, 1, 2, 6, 2, 6, 2)
    → def_scaling_factor(2, 1, 2, 6, 2, 6, 2)
"""
from .models import ScalingFactor, MotorCADGeo


def def_scaling_factor(
    k_radial: float,
    k_axial: float,
    define_type: int,
    *args,
) -> ScalingFactor:
    """스케일링 팩터 구조체를 생성한다.

    Parameters
    ----------
    k_radial    : 반경 방향 스케일 비율
    k_axial     : 축 방향 스케일 비율
    define_type : 0 ~ 3

    define_type=0  args = (k_winding,)
    define_type=1  args = (turns_per_coil, a_p)
                   k_winding은 미결정 — 이후 n_c_ref/a_p_ref로 별도 설정 필요
    define_type=2  args = (turns_per_coil, a_p, n_c_ref, a_p_ref)
                   k_winding = (turns_per_coil/a_p) / (n_c_ref/a_p_ref)
    define_type=3  args = (turns_per_coil, a_p, geo: MotorCADGeo)
                   n_c_ref / a_p_ref 를 geo에서 자동 취득

    Returns
    -------
    ScalingFactor
    """
    factor = ScalingFactor(k_radial=float(k_radial), k_axial=float(k_axial))

    if define_type == 0:
        if len(args) < 1:
            raise ValueError("define_type=0: args에 k_winding 필요")
        factor.k_winding = float(args[0])

    elif define_type == 1:
        if len(args) < 2:
            raise ValueError("define_type=1: args에 (turns_per_coil, a_p) 필요")
        factor.n_c = float(args[0])
        factor.a_p = float(args[1])
        # k_winding 미결정 — 호출자가 ref 정보와 함께 설정해야 함

    elif define_type == 2:
        if len(args) < 4:
            raise ValueError("define_type=2: args에 (turns_per_coil, a_p, n_c_ref, a_p_ref) 필요")
        turns_per_coil = float(args[0])
        a_p            = float(args[1])
        n_c_ref        = float(args[2])
        a_p_ref        = float(args[3])
        factor.n_c       = turns_per_coil
        factor.a_p       = a_p
        factor.k_winding = (turns_per_coil / a_p) / (n_c_ref / a_p_ref)

    elif define_type == 3:
        if len(args) < 3:
            raise ValueError("define_type=3: args에 (turns_per_coil, a_p, geo: MotorCADGeo) 필요")
        turns_per_coil = float(args[0])
        a_p            = float(args[1])
        geo: MotorCADGeo = args[2]
        n_c_ref = (geo.MagTurnsConductor if geo.Armature_CoilStyle == 0
                   else geo.WindingLayers)
        a_p_ref          = geo.ParallelPaths
        factor.n_c       = turns_per_coil
        factor.a_p       = a_p
        factor.k_winding = (turns_per_coil / a_p) / (n_c_ref / a_p_ref)

    else:
        raise ValueError(f"define_type은 0~3이어야 합니다. 입력값: {define_type}")

    return factor
